llms/mlx_lm/models/phi3.py

```diff
@@ -17,10 +17,10 @@ class ModelArgs(BaseModelArgs):
     num_attention_heads: int
     rms_norm_eps: float
     vocab_size: int
-    num_key_value_heads: int = None
+    num_key_value_heads: Optional[int] = None
     rope_theta: float = 10000
     rope_traditional: bool = False
-    rope_scaling: Optional[Dict[str, Union[float, str]]] = None
+    rope_scaling: Optional[Dict[str, Union[float, List[float]]]] = None
```

Comment:
Using Python dictionary implicitly presumed schemeless or typeless coding style, which is against the style of the rest fields in this dataclass. We may want to change this dictionary into a nested dataclass in the future.

Reply:
Indeed that would probably be better.

Reply:
I will try if that works today, then file another PR if it does.